### PR TITLE
IDS-696: update test data to be freely licensed

### PR DESCRIPTION
**Pull request by djc**
@nrempel not sure how much of this you want to review in detail, but please have a look. Also happy to get questions about where it's not obvious why/how something is done so I can clarify the code or add more comments.

### CODE REVIEW

File: instant-segment/examples/merge.rs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+fn read_word_list() -> HashSet<SmartString> {
+    const AVERAGE_WORD_LIST_LINE_LEN: usize = 9;
+
+    let f = File::open("data/cache/eng-wordlist.txt").unwrap();
+    let size = f.metadata().unwrap().len() as usize;
+    let mut reader = BufReader::with_capacity(4 * 1024 * 1024, f);
+
+    eprintln!("read word list...");
+    let mut word_list = HashSet::with_capacity(size / AVERAGE_WORD_LIST_LINE_LEN);
+    let mut ln = String::new();
```
Comment (nrempel) [inline instant-segment/examples/merge.rs:195]: Maybe add a comment here with what a line looks like in the data? I think that would make the parsing logic easier to grok.

File: instant-segment/examples/merge.rs [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        .map(|part| {
+            let fname = format!("data/cache/eng-2-{:05}-{:05}.txt", part, BIGRAM_PARTS);
+            let f = File::open(&fname).unwrap();
+            let mut reader = BufReader::with_capacity(4 * 1024 * 1024, f);
+
+            let mut ln = String::new();
+            let mut bigrams = HashMap::new();
+            loop {
+                ln.clear();
+                match reader.read_line(&mut ln) {
```
Comment (nrempel) [inline instant-segment/examples/merge.rs:32]: Could also give an example of what a line looks like here.

File: instant-segment/examples/merge.rs [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+                }
+
+                if bigrams.capacity() == 0 {
+                    // While it's not uncommon for a part to result in 0 words, the average for
+                    // parts that contain more than 0 is about 300k, median is about 350k. Allocate
+                    // a decent chunk immediately to avoid too many intermediate reallocations.
+                    bigrams.reserve(256 * 1024)
+                }
+
+                *bigrams.entry((word1, word2)).or_default() += matches;
```
Comment (nrempel) [inline instant-segment/examples/merge.rs:84]: As a Rust newbie, this immediately obvious what this is doing, can you explain?

**Review by nrempel — COMMENTED** [review]
Does it make sense to keep the data files out of git?

File: instant-segment/examples/merge.rs (hunk 3 above)
Comment (djc) [reply instant-segment/examples/merge.rs:84]: I'm using this: https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.or_default.

It's kind of like Python's `dict.setdefault()`. `entry()` returns an entry which can be `Vacant` or `Occupied`. `or_default()` will return a `&mut` to the value, inserting the `Default::default()` for the value's type if the entry is currently `Vacant`. `Default` is implemented for `usize` and yields 0.

File: instant-segment/examples/merge.rs (hunk 1 above)
Comment (djc) [reply instant-segment/examples/merge.rs:207]: Oh yeah, good idea. Added that.

File: instant-segment/examples/merge.rs (hunk 2 above)
Comment (djc) [reply instant-segment/examples/merge.rs:40]: Done.

**Comment by djc** [comment]
> Does it make sense to keep the data files out of git?

We had them in git before. I think we wanted to make them easy to get to for users, but I suppose we could do that some other way. What alternative way of hosting these files would you suggest?

File: instant-segment/examples/merge.rs (hunk 3 above)
Comment (nrempel) [reply instant-segment/examples/merge.rs:92]: Cool, thanks

**Comment by nrempel** [comment]
Are the files available from where you originally downloaded them? Could we download and cache the files on first run of the test? I don't think it's a horrible offence to have the data files checked in but it's nice to keep the git repo lean.

**Comment by djc** [comment]
The data files we have checked in after this PR are generated, so they are not available from anywhere. We could put them up for downloads somewhere (I suppose we can upload assets to GitHub somehow), but then having to download them for the tests adds substantial complexity (pulling in more dependencies, etc). It's 6 MB but at least it's text so it diffs well.

(BTW, we already have the current data files checked in, so this discussion is largely orthogonal to this PR.)

**Comment by nrempel** [comment]
Ah, yes in that case keeping the data in git seems like the best solution!

**Review by nrempel — APPROVED** [review]

**Pull request merged**